src/test/CdpRegistry.t.sol

```diff
@@ -38,54 +48,58 @@ contract CdpRegistryTest is TestBase {
+    Usr            usr1;
+    Usr            usr2;
 
     function setUp() public virtual {
         cdpManager  = new MockCdpManager();
         cdpRegistry = new CdpRegistry(address(cdpManager));
-        usr1        = address(new Usr());
-        usr2        = address(new Usr());
+        usr1        = new Usr(cdpRegistry);
+        usr2        = new Usr(cdpRegistry);
     }
 
-    function test_open_different_ilks() public {
-        uint256 cdp = cdpRegistry.open("ILK1", usr1);
+    function test_open_different_ilk_same_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr1));
         assertEq(cdp, 1);
         assertEq(cdpRegistry.ilks(1), "ILK1");
-        assertEq(cdpRegistry.owns(1), usr1);
-        assertEq(cdpRegistry.cdps("ILK1", usr1), 1);
+        assertEq(cdpRegistry.owns(1), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr1)), 1);
         assertEq(cdpManager.ilks(1), "ILK1");
         assertEq(cdpManager.owns(1), address(cdpRegistry));
 
-        cdp = cdpRegistry.open("ILK2", usr1);
+        cdp = usr1.open("ILK2", address(usr1));
         assertEq(cdp, 2);
         assertEq(cdpRegistry.ilks(2), "ILK2");
-        assertEq(cdpRegistry.owns(2), usr1);
-        assertEq(cdpRegistry.cdps("ILK2", usr1), 2);
+        assertEq(cdpRegistry.owns(2), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK2", address(usr1)), 2);
         assertEq(cdpManager.ilks(2), "ILK2");
         assertEq(cdpManager.owns(2), address(cdpRegistry));
     }
 
-    function test_open_different_users() public {
-        uint256 cdp = cdpRegistry.open("ILK1", usr1);
+    function test_open_same_ilk_different_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr1));
         assertEq(cdp, 1);
         assertEq(cdpRegistry.ilks(1), "ILK1");
-        assertEq(cdpRegistry.owns(1), usr1);
-        assertEq(cdpRegistry.cdps("ILK1", usr1), 1);
+        assertEq(cdpRegistry.owns(1), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr1)), 1);
         assertEq(cdpManager.ilks(1), "ILK1");
         assertEq(cdpManager.owns(1), address(cdpRegistry));
 
-        cdp = cdpRegistry.open("ILK1", usr2);
+        cdp = usr2.open("ILK1", address(usr2));
         assertEq(cdp, 2);
         assertEq(cdpRegistry.ilks(2), "ILK1");
-        assertEq(cdpRegistry.owns(2), usr2);
-        assertEq(cdpRegistry.cdps("ILK1", usr2), 2);
+        assertEq(cdpRegistry.owns(2), address(usr2));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr2)), 2);
         assertEq(cdpManager.ilks(2), "ILK1");
         assertEq(cdpManager.owns(2), address(cdpRegistry));
     }
 
-    function testFail_open_same_ilk_same_user() public {
-        uint256 cdp = cdpRegistry.open("ILK1", usr1);
-        cdp = cdpRegistry.open("ILK1", usr1);
+    function testFail_open_same_ilk_same_usr() public {
+        uint256 cdp = cdpRegistry.open("ILK1", address(usr1));
+        cdp = cdpRegistry.open("ILK1", address(usr1));
+    }
```

Comment:
This test should be the usr1 opening their own CDP, otherwise it will fail by the other require instead of the intended one.

Reply:
Fixed.

Reply:
Looks good.